src/lib/server/shop/getTickets.ts

```diff
@@ -41,7 +41,11 @@ const ticketIncludedFields = (id: DBShopIdentification) => ({
       consumables: {
         where: {
           ...id,
-          OR: [{ expiresAt: { gt: new Date() } }, { expiresAt: null }],
+          OR: [
+            { purchasedAt: { not: null } },
+            { expiresAt: { gt: new Date() } },
+            { expiresAt: null },
```

Comment:
Shouldn't it be something like `AND: [ { purchasedAt: { not: null } }, OR: [ { expiresAt: { gt: new Date() } }, { expiresAt: null }]]`? So that you have had to have purchased it? Or am I misunderstanding?

Reply:
It's supposed to get all items either in the user's cart (not purchased, not expired) OR items in the user's inventory (purchased)

So we don't really care about the purchase state, except that when `purchasedAt: null` there is another check, which does not exist when `purchasedAt: {not: null}`.

Reply:
Aah, thanks for clarifying. That makes sense.